operators/src/adapters/feature_collection_merger.rs

```diff
@@ -0,0 +1,250 @@
+use crate::util::Result;
+use futures::ready;
+use futures::stream::FusedStream;
+use futures::Stream;
+use geoengine_datatypes::collections::FeatureCollection;
+use geoengine_datatypes::primitives::Geometry;
+use geoengine_datatypes::util::arrow::ArrowTyped;
+use pin_project::pin_project;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// Merges a stream of `FeatureCollection` so that they are at least `chunk_byte_size` large.
+/// TODO: This merger outputs an empty stream if all collections are empty
+///     Do we need an empty collection with column info as output instead?
+///     Do we put the columns to the stream's `VectorQueryContext` instead?
+#[pin_project(project = FeatureCollectionChunkMergerProjection)]
+pub struct FeatureCollectionChunkMerger<St, G>
+where
+    St: Stream<Item = Result<FeatureCollection<G>>> + FusedStream,
+    G: Geometry + ArrowTyped,
+{
+    #[pin]
+    stream: St,
+    accum: Option<FeatureCollection<G>>,
+    chunk_size_bytes: usize,
+}
+
+impl<St, G> FeatureCollectionChunkMerger<St, G>
+where
+    St: Stream<Item = Result<FeatureCollection<G>>> + FusedStream,
+    G: Geometry + ArrowTyped,
+{
+    pub fn new(stream: St, chunk_size_bytes: usize) -> Self {
+        Self {
+            stream,
+            accum: None,
+            chunk_size_bytes,
+        }
+    }
+
+    fn merge_and_proceed(
+        accum: &mut Option<FeatureCollection<G>>,
+        chunk_size_bytes: usize,
+        new_collection: St::Item,
+    ) -> Option<Poll<Option<St::Item>>> {
+        if new_collection.is_err() {
+            // TODO: maybe first output existing chunk and then the error?
+            return Some(Poll::Ready(Some(new_collection)));
+        }
+
+        let new_collection = new_collection.expect("checked");
+
+        let merged_collection = if let Some(old_collection) = accum.take() {
+            // TODO: execute on separate thread?
+            old_collection.append(&new_collection)
+        } else {
+            Ok(new_collection)
+        };
+
+        match merged_collection {
+            Ok(collection)
+                if !collection.is_empty() && collection.byte_size() >= chunk_size_bytes =>
```

Comment:
computing the byte size is not for free. maybe store the current acum byte size and add the byte size of the added chunk to it instead of recomputing the total byte size over and over again

Reply:
I guess calculating the byte size is not more expensive for collections with more rows, since it depends on the number of buffers (attributes).
This means, calculating the size on the merged collection equals calculating it on the collection to append and summing the byte sizes.

The remaining question is, is the same byte size calculated multiple times?
I say no since there are two scenarios:
1. The collection is large enough and out goes the collection as well as the validity of the calculated byte size.
2. The collection is not large enough. Then it gets merged and a new byte size is the valid one which must be computed no matter what.
Hence, I doubt that it will save us computation time to store it in the merger.

Reply:
I think it should be more efficient to calculate the byte_size of the individual `new_collection`s and add them up as they come. Because the way you do it you have to compute the size of the first merged chunk again every time a new chunk is merged. If you have a chunk_size of e.g. 100MB and the underlying stream produces chunks of 1 MB this might be already significant.

Of course that all only makes sense if byte_size(collection1 + collection2) = byte_size(collection1) + byte_size(collection2)

Reply:
Could you re-check that the calculation is row-bound? My previous post doubted that since if sums up the size of the buffers.